drivers/flash/flash_simulator.c

```diff
@@ -174,9 +174,8 @@ static int flash_range_is_valid(const struct device *dev, off_t offset,
 				size_t len)
 {
 	ARG_UNUSED(dev);
-	if ((offset + len > FLASH_SIMULATOR_FLASH_SIZE +
-			    FLASH_SIMULATOR_BASE_OFFSET) ||
-	    (offset < FLASH_SIMULATOR_BASE_OFFSET)) {
+
+	if ((offset + len > FLASH_SIMULATOR_FLASH_SIZE) || (offset < 0)) {
```

Comment:
Since you are fixing this, can we fix the somehow absurd check for correctness of `offset` _after_ doing calculations with it? I know that it was here before...
Also the more correct approach would be to do:
```
(offset < 0 || offset >= FLASH_SIMULATOR_FLASH_SIZE || (FLASH_SIMULATOR_FLASH_SIZE - offset) < size)
```
this should let us avoid all overflows the addition could cause.

Reply:
Good catch! Fixed. BTW: The reason, why I'm fixing this is #79130.